exchanges/okcoin/okcoin_websocket.go

```diff
@@ -147,584 +247,690 @@ func (o *OKCoin) WsHandleData(respRaw []byte) error {
-	switch num {
-	case -2:
-		return order.Rejected, nil
-	case -1:
-		return order.Cancelled, nil
-	case 0:
-		return order.Active, nil
-	case 1:
-		return order.PartiallyFilled, nil
-	case 2:
-		return order.Filled, nil
-	case 3:
-		return order.New, nil
-	case 4:
-		return order.PendingCancel, nil
-	default:
-		return order.UnknownStatus, fmt.Errorf("%v not recognised as order status", num)
-	}
-}
-
-func (o *OKCoin) wsProcessOrder(respRaw []byte) error {
-	var resp WebsocketSpotOrderResponse
+func (o *Okcoin) wsProcessAdvancedAlgoOrder(respRaw []byte) error {
+	var resp WebsocketAdvancedAlgoOrder
 	err := json.Unmarshal(respRaw, &resp)
 	if err != nil {
 		return err
 	}
-	for i := range resp.Data {
-		var oType order.Type
-		var oSide order.Side
-		var oStatus order.Status
-		oType, err = order.StringToOrderType(resp.Data[i].Type)
+	algoOrders := make([]order.Detail, len(resp.Data))
+	for d := range resp.Data {
+		oType, err := order.StringToOrderType(resp.Data[d].OrderType)
 		if err != nil {
-			o.Websocket.DataHandler <- order.ClassificationError{
-				Exchange: o.Name,
-				OrderID:  resp.Data[i].OrderID,
-				Err:      err,
-			}
+			return err
 		}
-		oSide, err = order.StringToOrderSide(resp.Data[i].Side)
+		oSide, err := order.StringToOrderSide(resp.Data[d].Side)
 		if err != nil {
-			o.Websocket.DataHandler <- order.ClassificationError{
-				Exchange: o.Name,
-				OrderID:  resp.Data[i].OrderID,
-				Err:      err,
-			}
+			return err
 		}
-		oStatus, err = StringToOrderStatus(resp.Data[i].State)
+		// this code block introduces two order states
+		// 1. 'effective' - equivalent to filled, and
+		// 2. 'order_failed' == equivalent to failed
+		oStatus, err := order.StringToOrderStatus(resp.Data[d].State)
 		if err != nil {
-			o.Websocket.DataHandler <- order.ClassificationError{
-				Exchange: o.Name,
-				OrderID:  resp.Data[i].OrderID,
-				Err:      err,
-			}
+			return err
 		}
-
-		pair, err := currency.NewPairFromString(resp.Data[i].InstrumentID)
+		cp, err := currency.NewPairFromString(resp.Data[d].InstrumentID)
 		if err != nil {
-			o.Websocket.DataHandler <- order.ClassificationError{
-				Exchange: o.Name,
-				OrderID:  resp.Data[i].OrderID,
-				Err:      err,
-			}
-		}
-
-		o.Websocket.DataHandler <- &order.Detail{
-			ImmediateOrCancel: resp.Data[i].OrderType == 3,
-			FillOrKill:        resp.Data[i].OrderType == 2,
-			PostOnly:          resp.Data[i].OrderType == 1,
-			Price:             resp.Data[i].Price,
-			Amount:            resp.Data[i].Size,
-			ExecutedAmount:    resp.Data[i].LastFillQty,
-			RemainingAmount:   resp.Data[i].Size - resp.Data[i].LastFillQty,
-			Exchange:          o.Name,
-			OrderID:           resp.Data[i].OrderID,
-			Type:              oType,
-			Side:              oSide,
-			Status:            oStatus,
-			AssetType:         o.GetAssetTypeFromTableName(resp.Table),
-			Date:              resp.Data[i].CreatedAt,
-			Pair:              pair,
+			return err
```

Comment:
Instead of using `return err` here, could you use something like:
```go
	b.Websocket.DataHandler <- order.ClassificationError{
					Exchange: b.Name,
					OrderID:  notification.Data[i].OrderID,
					Err:      err,
				}
```
This is mostly because rather than returning an error for order 1 of 10 (for example) its ultimately best to know about orders, even if they don't have the correct order type (for example) . At least then we can query the order via orderID and get the right info